nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -105,6 +105,9 @@ void RegulatedPurePursuitController::configure(
     node, plugin_name_ + ".max_angular_accel", rclcpp::ParameterValue(3.2));
   declare_parameter_if_not_declared(
     node, plugin_name_ + ".allow_reversing", rclcpp::ParameterValue(false));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".max_distance_between_iterations",
```

Comment:
Note: I myself am terrible with naming things. I don't like this variable name, its not really very descriptive without reading alot of documentation or the code itself.

How about: `max_robot_pose_search_dist`? 

I think either way we'll need a healthy few sentences in documentation, but at least then describes "why" this is here and what it might be doing. I would recommend when we get to docs making sure to note that we do not recommend changing this from the default unless you have full loops or criss-crosses within your local costmap size, which you may then need to reduce proportionally to those sizes

Reply:
Yeah I think `max_robot_pose_search_dist` is just fine. I can change it to that.

Reply:
Where would be the best place to put those docs? In the README.md for `nav2_regulated_pure_pursuit_controller`?

Reply:
Yeah, there should be a table with parameters in the readme you can add too. Also, once this is done, we'll need to update docs on navigation.ros.org that would also reflect it